File: php/24_Swap_Nodes_in_Pairs.php
<?php

/**
 * Definition for a singly-linked list.
 * class ListNode {
 *     public $val = 0;
 *     public $next = null;
 *     function __construct($val = 0, $next = null) {
 *         $this->val = $val;
 *         $this->next = $next;
 *     }
 * }
 */
class Solution
{

    /**
     * @param ListNode $head
     * @return ListNode
     */
    function swapPairs($head)
    {
        $temp = new ListNode(0);
        $temp->next = $head;
        $current = $temp;
        while ($current->next != null && $current->next->next != null) {
            $slow = $current->next;
            $fast = $current->next->next;
            $slow->next = $fast->next;
            $current->next = $fast;
            $current->next->next = $slow;
            $current = $current->next->next;
        }

        return $temp->next;
    }

}

File: go/24_Swap_Nodes_in_Pairs.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func swapPairs(head *ListNode) *ListNode {
    temp := new (ListNode);
    temp.Next = head;
    current := temp;
    for current.Next != nil && current.Next.Next != nil {
        slow := current.Next;
        fast := current.Next.Next;
        slow.Next = fast.Next;
        current.Next = fast;
        current.Next.Next = slow;
        current = current.Next.Next;
    }

    return temp.Next;
}
